game-core/src/main/java/games/strategy/triplea/delegate/BattleTracker.java

```diff
@@ -81,9 +81,8 @@
   private BattleRecords battleRecords = null;
   // to keep track of all relationships that have changed this turn
   // (so we can validate things like transports loading in newly created hostile zones)
-  private final Collection<
-      Tuple<Tuple<PlayerId, PlayerId>, Tuple<RelationshipType, RelationshipType>>> relationshipChangesThisTurn =
-          new ArrayList<>();
+  private final Collection<Tuple<Tuple<PlayerId, PlayerId>, Tuple<RelationshipType, RelationshipType>>> relationshipChangesThisTurn =
```

Comment:
I dunno. I re-imported again and doesn't seem to matter. Maybe its different versions of eclipse or something?

Reply:
I think we are still missing a setting in the formatter. Need to have:
Line Wrapping > Statements > 'for' > Line Wrapping Policy > Wrap where necessary

Reply:
> Maybe its different versions of eclipse or something?

IIRC, based on the exported formatter settings you attached to #4123, I'm pretty sure I generated the latest formatter configuration on `master` using Oxygen (as you said that's what you were using at the time).  Are you still using Oxygen (4.7)?  Or have you upgraded to Photon (4.8) or 2018-09 (4.9)?

> I think we are still missing a setting in the formatter. Need to have:
Line Wrapping > Statements > 'for' > Line Wrapping Policy > Wrap where necessary

It's possible, but the affected `for` statement in `BattleTracker` seems to wrap correctly for me in Photon with the current configuration.  Allowing line breaks between type parameters et al., as enabled in #4136, should have an equivalent effect.  Can you verify that all (3) child nodes under **Parameterized types** are set to **Wrap where necessary**?

Reply:
@ssoloff Yep, all 3 under Parameterized types are set to Wrap where necessary.

Reply:
> I'm thinking we should probably have pretty much everything as 'wrap where necessary'.

No objection.  However, we should verify that doesn't reformat a ton of code.  I suspect it won't since we don't currently have any long lines, but it would be prudent to do that before committing to the new configuration.

> And yeah, I'm still on 4.7.

I've been running 4.8 for about three months, and I've since uninstalled 4.7, so I can't test the formatter config any longer in Oxygen to see if I can reproduce the problem.

I'm planning on upgrading to 4.9 next month.  Whatever change we make, as discussed above, we should test it on all the Eclipse platforms we plan on using.  If you submit a PR and test on one version, I can test it before merging on whatever version I'm using (if different).